neo/core/imagesequence.py

```diff
@@ -0,0 +1,196 @@
+        :name: (str) A label for the dataset.
+        :description: (str) Text description.
+        :file_origin: (str) Filesystem path or URL of the original data file.
+
+    *Optional attributes/properties*:
+        :dtype: (numpy dtype or str) Override the dtype of the signal array.
+        :copy: (bool) True by default.
+
+    Note: Any other additional arguments are assumed to be user-specific
+    metadata and stored in :attr:`annotations`.
+
+    *Properties available on this object*:
+        :sampling_rate: (quantity scalar) Number of samples per unit time.
+            (1/:attr:`sampling_period`)
+        :sampling_period: (quantity scalar) Interval between two samples.
+            (1/:attr:`quantity scalar`)
+        :spatial_scale: size of a pixel
+     """
+    _single_parent_objects = ('Segment',)
+    _single_parent_attrs = ('segment',)
+    _quantity_attr = 'image_data'
+    _necessary_attrs = (('image_data', pq.Quantity, 3),
+                        ('sampling_rate', pq.Quantity, 0),
+                        ('spatial_scale', pq.Quantity, 0))
+    _recommended_attrs = BaseNeo._recommended_attrs
+
+    def __new__(cls, image_data, units=None, dtype=None, copy=True, spatial_scale=None, sampling_period=None,
+                sampling_rate=None, name=None, description=None, file_origin=None,
+                **annotations):
+        """
+        Constructs new :class:`ImageSequence` from data.
+
+        This is called whenever a new class:`ImageSequence` is created from
+        the constructor, but not when slicing.
+
+        __array_finalize__ is called on the new object.
+        """
+        if spatial_scale is None:
+            raise ValueError('spatial_scale is required')
+
+        image_data = np.stack(image_data)
+        if len(image_data.shape) != 3:
+            raise ValueError('list doesn\'t have the good number of dimension')
+
+        obj = pq.Quantity(image_data, units=units, dtype=dtype, copy=copy).view(cls)
+        obj.segment = None
+        # function from analogsignal.py in neo/core directory
+        obj.sampling_rate = _get_sampling_rate(sampling_rate, sampling_period)
+        obj.spatial_scale = spatial_scale
+
+        return obj
+
+    def __init__(self, image_data, units=None, dtype=None, copy=True, spatial_scale=None, sampling_period=None,
+                 sampling_rate=None, name=None, description=None, file_origin=None,
+                 **annotations):
+        '''
+               Initializes a newly constructed :class:`ImageSequence` instance.
+        '''
+        DataObject.__init__(self, name=name, file_origin=file_origin, description=description,
+                            **annotations)
+
+    def __array_finalize__spec(self, obj):
+
+        self.sampling_rate = getattr(obj, 'sampling_rate', None)
+        self.spatial_scale = getattr(obj, 'spatial_scale', None)
+        self.units = getattr(obj, 'units', None)
+
+        return obj
+
+    def signal_from_region(self, *region):
```

Comment:
I think we would a **reduce_mode='average' / 'median'/'max'** kargs to control the method of the reduce.
No ?

Reply:
yes, that would be a good improvement. I suggest leaving it for a future update, however.